src/main/scala/viper/gobra/translator/encodings/channels/ChannelEncoding.scala

```diff
@@ -0,0 +1,248 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.encodings.channels
+
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.reporting.Source
+import viper.gobra.theory.Addressability
+import viper.gobra.theory.Addressability.{Exclusive, Shared}
+import viper.gobra.translator.Names
+import viper.gobra.translator.encodings.LeafTypeEncoding
+import viper.gobra.translator.interfaces.Context
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class ChannelEncoding extends LeafTypeEncoding {
+
+  import viper.gobra.translator.util.TypePatterns._
+  import viper.gobra.translator.util.ViperWriter.CodeLevel._
+
+  /**
+    * Translates a type into a Viper type.
+    */
+  override def typ(ctx : Context) : in.Type ==> vpr.Type = {
+    case ctx.Channel(t) / m =>  m match {
+      case Exclusive => vpr.Int // for now we simply map it to Viper Ints
+      case Shared => vpr.Ref
+    }
+  }
+
+  /**
+    * Encodes expressions as values that do not occupy some identifiable location in memory.
+    *
+    * R[ dflt(channel[T]) ] -> 0
+    * R[ nil : channel[T] ] -> 0
+    * R[ <- (c: channel[T] ] ->
+    *   exhale acc([c].RecvChannel(), wildcard)
+    *   exhale [c].RecvGivenPerm()()
+    *   var res [ T ]
+    *   inhale [c].RecvChannel()
```

Comment:
I am surprised that `res` does not occur anywhere else.

Reply:
This read operation is indeed pretty useless but I support it for completeness reasons. 
I basically use the same contract as for the safe receive operation but since the success flag is missing all implications in the postcondition can not be expressed. Hence, an arbitrary value (incl. the zero value) is received and one cannot distinguish receiving a ‚valid‘ value and receiving from a closed channel

Reply:
But then there are restrictions on res, it is a zero value or a value that satisfies the channel invariant. Therefore, there should be an assumption on res.

Reply:
I've now added the following inhale making the encoding slightly stronger:
`inhale res != Dflt[T] ==> [c].RecvGotPerm()(res)`
However as Martin pointed out, he would maybe even go as far as simply disallowing this form of receive